changelog.d/11395.removal

```diff
@@ -0,0 +1 @@
+Remove code related to deprecated 'trust_identity_servers_for_password_resets' config flag.
```

Comment:
I would probably use the same changelog text as the other PR so that they get merged together. As it stands, this will be a separate entry in the 'removals' section in the changelog and it won't really make much sense to homeserver admins what the legacy code is that we speak of — and I'm not sure it needs to.